src/lib/run-study.ts

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable no-console */
+import {
+    ECSClient,
+    LaunchType,
+    RunTaskCommand,
+    RunTaskCommandInput,
+    RegisterTaskDefinitionCommand,
+    RegisterTaskDefinitionCommandInput,
+    DescribeTaskDefinitionCommand,
+    DescribeTaskDefinitionCommandOutput,
+    RunTaskCommandOutput,
+} from '@aws-sdk/client-ecs'
+import { GetResourcesCommand, ResourceGroupsTaggingAPIClient } from '@aws-sdk/client-resource-groups-tagging-api'
+
+async function launchStudy(
+    client: ECSClient,
+    cluster: string,
+    baseTaskDefinition: string,
+    subnets: string,
+    securityGroup: string,
+    studyId: string,
+    studyImage: string,
+): Promise<RunTaskCommandOutput> {
+    const baseTaskDefinitionData = await getTaskDefinition(client, cluster, baseTaskDefinition)
+    const containerDefinition = baseTaskDefinitionData.taskDefinition?.containerDefinitions?.map((container) => {
+        return {
+            ...container,
+            image: studyImage,
+        }
+    })
+
+    // Create a derived task definition for the study using the base
+    // The following defines a new family versus versioning the base family
```

Comment:
This notion of a base task definition is based upon the idea that there is value in having the bulk of the "source of truth" [live in our IaC](https://github.com/safeinsights/iac/blob/fbcb174d1106d2729a894f0269dad8f52236808a/secure-enclave/stack.ts#L246-L260) versus application code. Maybe that turns out to be a bad idea later, but for now it allows us to keep the code here pretty simple I think by avoiding having to define the various roles and permissions (like access to secrets in the execution role that CDK automagically does for us), etc. :man_shrugging: .